plugins/keep-markup/tests/test.js

```diff
@@ -0,0 +1,88 @@
+const expect = require('chai').expect;
+const jsdom = require('jsdom')
+const { JSDOM } = jsdom
+
+require('../../../prism')
+// fake DOM
+global.self = {}
+global.self.Prism = Prism
+global.document = {}
+document.createRange = function () {
+}
+global.self.document = document
+
+require('../prism-keep-markup')
+
+describe('Prism Keep Markup Plugin', function () {
+
+	function execute (code) {
+		const start = [];
+		const end = [];
+		const nodes = [];
+		document.createRange = function () {
+			return {
+				setStart: function (node, offset) {
+					start.push({ node, offset })
+				},
+				setEnd: function (node, offset) {
+					end.push({ node, offset })
+				},
+				extractContents: function () {
+					return new JSDOM('').window.document.createTextNode('')
+				},
+				insertNode: function (node) {
+					nodes.push(node)
+				},
+				detach: function () {
+				}
+			}
+		}
+		const beforeHighlight = Prism.hooks.all['before-highlight'][0]
+		const afterHighlight = Prism.hooks.all['after-highlight'][0]
+		const env = {
+			element: new JSDOM(code).window.document.getElementsByTagName('code')[0],
+			language: "javascript"
+		}
+		beforeHighlight(env)
+		afterHighlight(env)
+		return { start, end, nodes }
+	}
+
+	it('should keep <span> markup', function () {
+		const result = execute(`<code class="language-javascript">x<span>a</span>y</code>`)
+		expect(result.start.length).to.equal(1)
+		expect(result.end.length).to.equal(1)
+		expect(result.nodes.length).to.equal(1)
+		expect(result.nodes[0].nodeName).to.equal('SPAN')
+	})
+	it('should preserve markup order', function () {
```

Comment:
This test currently fails, `<a></a><b></b>` are inverted in the result.

**Input**
```html
<code class="language-javascript">x<a></a><b></b>y</code>
```
**Output**
```html
<code class="language-javascript">x<b></b><a></a>y</code>
```